### ISSUE TITLE
arm64 support

### ISSUE THREAD

**Issue by P-AS** [clementgallet/libTAS#487]
Hello, since my main computer is a Macbook Pro with M1, it would be very convenient to use libTAS on this computer without needing to carry around another laptop just to use libTAS. It would have less applications than x86, but you could still be able to TAS a lot with it.

**Comment by InfoTeddy** [comment]
Have you tried compiling for ARM?

**Comment by clementgallet** [comment]
There are a few things that would need to be implemented for ARM, but not much. I don't have a machine so I cannot test it. Did you try compiling it?

**Comment by PandaSRC** [comment]
This is what happens when you try to compile it
[libtasarm.txt](https://github.com/clementgallet/libTAS/files/8776417/libtasarm.txt)

**Comment by InfoTeddy** [comment]
Yeah, that seems about right. Pull requests for ARM are accepted if you want to do the work, or you can wait for Kilaye or someone else to do it.

**Comment by InfamousKnight** [comment]
There's an emulator called box64 you can try as well.

**Comment by P-AS** [comment]
On macOS Ventura you can install rosetta on linux inside a vm ([video](https://youtu.be/BVrxyszJ37I)), which does let you run libTAS on arm64. However, rosetta prevents you from loading savestates.
<img width="1392" alt="rosettavm" src="https://user-images.githubusercontent.com/43502756/180957904-80d2af64-d7c6-40e5-8ce6-46362765ae4f.png">
Testing with ruffle:
```
        not a dynamic executable
Attempt 1: Connected.
[libTAS f:69] Thread 17590 (main) Saved state 1 of size 74711566 in 0.368040 seconds
rosetta error: futex(FUTEX_LOCK_PI_PRIVATE) failure: 35
recv() returns 0 -> socket closed
Got wrong message after state loading
Trace/breakpoint trap
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 1648, resource id: 14706942, major code: 40 (TranslateCoords), minor code: 0```

**Comment by P-AS** [comment]
After years of no progress on this, I gave in and used Claude. 7f94d34d97ae247c4afda878bfd7c8945a8300bd (I did try to do this by hand before, I couldn't do it.)
It's fully functional, games run, movies made on x86_64 sync, savestates work, and on non-4k page size kernels too. You could easily make an entire TAS with this if your game/emulator is supported.
Since it's LLM generated, I don't fully understand the code, and I don't want to potentially harm libTAS, I'm not going to open a pull request for this. A human solution would still be preferred, even if it uses this as a reference.

**Comment by clementgallet** [comment]
Yeah, assuming 4096-byte page size is bad code.

Most of it looks okay to code, or to fetch from the project it was taken from (there's a bit of CRIU code). The most brutal part is the function patching, where we need to relocate the first instructions of a function into a new memory segment, and modify the instructions so that the addresses are valid. This took me tens of hours digging into the x86 asm specs. aarch64 has a much more manageable instruction set, but it would still require a lot of work.

However, function patching is currently only necessary for Unity games or wine, so it doesn't need to be implemented for aarch64 I think

It looks like qemu can emulate an aarch64 CPU easily, so I could test myself eventually
